src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingServiceTest.kt

```diff
@@ -0,0 +1,185 @@
+
+            // Then
+            verify(printDetailsRepository).getAllByStatusAndBatchId(SENT_TO_PRINT_PROVIDER, batchResponse1.batchId)
+            verify(printDetailsRepository).getAllByStatusAndBatchId(SENT_TO_PRINT_PROVIDER, batchResponse2.batchId)
+            verify(printDetailsRepository, times(2)).updateItems(capture(captor))
+            assertThat(captor.allValues[0]).usingRecursiveComparison().isEqualTo(listOf(expectedPrintDetails1))
+            assertThat(captor.allValues[1]).usingRecursiveComparison().isEqualTo(listOf(expectedPrintDetails2))
+            verify(idFactory).requestId()
+        }
+    }
+
+    @Nested
+    inner class ProcessPrintResponse {
+        @ParameterizedTest
+        @CsvSource(
+            value = [
+                "RECEIVED_BY_PRINT_PROVIDER, PROCESSED, SUCCESS, VALIDATED_BY_PRINT_PROVIDER",
+                "VALIDATED_BY_PRINT_PROVIDER, IN_MINUS_PRODUCTION, SUCCESS, IN_PRODUCTION",
+                "IN_PRODUCTION, DISPATCHED, SUCCESS, DISPATCHED",
+                "DISPATCHED, NOT_MINUS_DELIVERED, FAILED, NOT_DELIVERED",
+                "RECEIVED_BY_PRINT_PROVIDER, PROCESSED, FAILED, PRINT_PROVIDER_VALIDATION_FAILED",
+                "VALIDATED_BY_PRINT_PROVIDER, IN_MINUS_PRODUCTION, FAILED, PRINT_PROVIDER_PRODUCTION_FAILED  ",
+                "IN_PRODUCTION, DISPATCHED, FAILED, PRINT_PROVIDER_DISPATCH_FAILED"
+            ]
+        )
+        fun `should update print details`(
+            initialStatus: Status,
+            statusStep: ResponseStatusStep,
+            status: ResponseStatus,
+            expectedStatus: Status
+        ) {
+            // Given
+            val printDetails = buildPrintDetails(status = initialStatus)
+            val requestId = printDetails.requestId!!
+            given(printDetailsRepository.getByRequestId(any())).willReturn(printDetails)
+            val response = buildProcessPrintResponseMessage(
+                requestId = requestId,
+                statusStep = statusStep,
+                status = status
+            )
+            val expectedPrintRequestStatus = PrintRequestStatus(
+                status = expectedStatus,
+                dateCreated = now,
+                eventDateTime = response.timestamp,
+                message = response.message
+            )
+            val expectedPrintDetails = printDetails.copy().apply {
+                printRequestStatuses = printRequestStatuses!!.toMutableList().apply { add(expectedPrintRequestStatus) }
+            }
+            // When
+            service.processPrintResponse(response)
+
+            // Then
+            verify(printDetailsRepository).getByRequestId(requestId)
+            verify(printDetailsRepository).updateItems(capture(captor))
+            assertThat(captor.value).usingRecursiveComparison().isEqualTo(listOf(expectedPrintDetails))
+        }
+    }
+
+    @Test
+    fun `should throw exception given non defined statusStep and status combination`() {
```

Comment:
This test can be renamed to 
`should throw exception given status is success and statusStep is not-delivered`

I cannot think of any other combination where an exception will be thrown

Reply:
That is true @vishalgupta-valtech 👍 

Though I think both of these tests (this one, and the main one above that does all the combinations), can be moved to a mapper test if we can implement the logic in a mapper 🤞